packages/quantic/force-app/main/default/lwc/quanticRefineToggle/quanticRefineToggle.js

```diff
@@ -0,0 +1,207 @@
+import {
+  initializeWithHeadless,
+  registerComponentForInit,
+} from 'c/quanticHeadlessLoader';
+import LOCALE from '@salesforce/i18n/locale';
+import sortAndFilters from '@salesforce/label/c.quantic_SortAndFilters';
+import viewResults from '@salesforce/label/c.quantic_ViewResults';
+
+/**
+ * @typedef {Object} QuanticModalElement
+ * @method openModal
+ * @method closeModal
+ */
+
+/** @typedef {import("coveo").SearchEngine} SearchEngine */
+/** @typedef {import("coveo").QuerySummary} QuerySummary */
+/** @typedef {import("coveo").BreadcrumbManager} BreadcrumbManager */
+
+/**
+ * The `QuanticRefineToggle` component displays a button that is used to open the refine modal.
+ * @category Search
+ * @example
+ * <c-quantic-refine-toggle engine-id={engineId} hide-icon button-label="Custom label" hide-sort full-screen>
+ *   <div slot="refine-title">Custom Title</div>
+ * </c-quantic-refine-toggle>
+ */
+export default class QuanticRefineToggle extends LightningElement {
+  labels = {
+    sortAndFilters,
+    viewResults,
+  };
+
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * Indicates whether the icon of the refine toggle button should be hidden.
+   * @api
+   * @type {string}
+   */
+  @api hideIcon;
+  /**
+   * The label to be shown in the refine toggle button.
+   * @api
+   * @type {string}
+   * @defaultValue `'Sort & Filters'`
+   */
+  @api buttonLabel = this.labels.sortAndFilters;
+  /**
+   * Indicates whether the Quantic Sort component should be hidden.
+   * @api
+   * @type {boolean}
+   */
+  @api hideSort;
+  /**
+   * Indicates whether the refine modal will be opened in full screen.
+   * @api
+   * @type {boolean}
+   * @defaultValue `false`
+   */
+  @api fullScreen = false;
+
+  /** @type {QuerySummary} */
+  querySummary;
+  /** @type {BreadcrumbManager} */
+  breadcrumbManager;
+  /** @type {number} */
+  total;
+  /** @type {number} */
+  activeFiltersCount = 0;
+  /** @type {string} */
+  modalId = 'refineModal';
+
+  connectedCallback() {
+    registerComponentForInit(this, this.engineId);
+  }
+
+  renderedCallback() {
+    initializeWithHeadless(this, this.engineId, this.initialize);
+  }
+
+  /**
+   * @param {SearchEngine} engine
+   */
+  initialize = (engine) => {
+    this.querySummary = CoveoHeadless.buildQuerySummary(engine);
+    this.breadcrumbManager = CoveoHeadless.buildBreadcrumbManager(engine);
+
+    this.unsubscribeQuerySummary = this.querySummary.subscribe(() =>
+      this.updateTotalResults()
+    );
+    this.unsubscribeBreadcrumbManager = this.breadcrumbManager.subscribe(() =>
+      this.updateActiveFiltersCount()
+    );
+  };
+
+  disconnectedCallback() {
+    this.unsubscribeQuerySummary?.();
+    this.unsubscribeBreadcrumbManager?.();
+  }
+
+  /**
+   * Updates total number of results.
+   * @returns {void}
+   */
+  updateTotalResults() {
+    this.total = this.querySummary.state.total;
```

Comment:
Because with this updateTotalResults method, I'm always listening to the state changes of the Query summary state and updating the right value of `total`, not sure if with a getter that returns this.querySummary.state.total, the component will re-render when the `total` value in the state changes 🧐

Reply:
getters are reactive in LWC afaik but you make valid points so keep as is 👍

Reply:
Yes getters are reactive, but it means LWC would need to track the whole QuerySummary state, which I don't think it would do automatically. That's the reason why the `@track` annotation has to be added when the attribute is an object or an array.